src/datapath.jl

```diff
@@ -80,6 +89,54 @@ function TSdataset(filepath::String)
     end
 end
 
+"""
+   `load_ts_file(fpath)`
+`load_ts_file` takes the path to `ts` files and returns the table (IndexedTable)
+and Array, where `fpath` is path to `ts` file located on your computer.
+eg
+```julia
+   table_a, arr = load_ts_file("/Users/your_user_name/../GunPoint_TRAIN.ts")
+```
+"""
+function load_ts_file(fpath)
+#`readuntil` and `readlines` are used in combination so that we read data
+# required only for transformation into julia array and table.
+    data = open(fpath) do input
+          readuntil(input, "@data")
+          readlines(input)
+    end
+    data = data[2:end] # removes  the empty string.
+    # split the string and convert each element into Float64.
+    arrays = map(i -> parse.(Float64, split(i, r"[:,]")) , data)
+    array = transpose(hcat(arrays...))*1 # Creates 2D Array.
+    return table(eachcol(array[:, 1:end-1])...), table(array[:, end])
+end
+
+"""
+   `ts_dataset(dataset, test_or_train)`
+`ts_dataset` takes two inputes, `dataset` is the name of datasets available in the
+`data` folder & `test_or_train` specifile the train or test part.
+Reurns dataset and target variable.
+eg
+```julia
+   X, y = ts_dataset("Adiac", nothing)
+```
+"""
+function ts_dataset(dataset::String, split=nothing)
+    if split in ["test", "train"]
+         fpath = joinpath(DATA_DIR, dataset, string(dataset, "_", uppercase(split), ".ts" ))
+         return load_ts_file(fpath)
+    elseif split == nothing
+         v = []
+         for i in ["test", "train"]
+             fpath = joinpath(DATA_DIR, dataset, string(dataset, "_", uppercase(i), ".ts" ))
+             push!(v, fpath)
+         end
+         test, train =  load_ts_file.(v)
+         return merge(test[1], train[1]), merge(test[2], train[2])
+    end
```

Comment:
add else clause which raises error if split is not one of nothing, train, test

Reply:
Added the error massage.